PWGHF/vertexingHF/AliAnaTaskSEXic0SL.C

```diff
@@ -0,0 +1,947 @@
+	if (event->IsA() == AliAODEvent::Class()) fEvt = dynamic_cast<AliAODEvent*>(event);
+	else { Printf("ERROR: input event type is NOT AOD"); return; } //Return 1
+
+	//Check run number
+	fEvtRunNo = fEvt->GetRunNumber();
+	if (fEvtRunNo<cut_runNoLo || fEvtRunNo>cut_runNoUp) { Printf("ERROR: runNo"); return; } //Return 2
+	fHisto->FillTH1("RunNo", fEvtRunNo, 1); //@
+
+	//Check bfield
+	const Double_t t_bfield = (Double_t)fEvt->GetMagneticField();
+	if (fabs(t_bfield) < cut_bfield) { Printf("ERROR: B-field"); return; } //Return 3
+	fHisto->FillTH1("EvtCut", "Bfield", 1); //@
+
+	//Check multiplicity selection is available
+	fMultSel = (AliMultSelection*)fEvt->FindListObject("MultSelection");
+	if (!fMultSel) { Printf("ERROR: MultSelection"); return; } //Return 4
+	else fEvtMult = fMultSel->GetMultiplicityPercentile(IsPA?"V0A":"V0M");
+	fHisto->FillTH1("EvtCut", "Mult", 1); //@
+
+	//Event validity (?)
+	fInputHandler = (AliInputEventHandler*)AliAnalysisManager::GetAnalysisManager()->GetInputEventHandler();
+	if (fInputHandler->IsEventSelected() == false) return; //Return 5
+
+	//Check pID
+	fPID = (AliPIDResponse*)fInputHandler->GetPIDResponse();
+	if (!fPID) { Printf("ERROR: PID"); return; } //Return 6
+	fHisto->FillTH1("EvtCut", "IsSel+PID", 1); //@
+
+	//Separate data/MC
+	if (IsMC == false) //data
+	{
+		//Check trigger
+		bool t_trigFired = false;
+		fEvtTrig = fInputHandler->IsEventSelected();
+		for (unsigned int a=0; a<TrigStore.size(); a++)
+		{
+			if (fEvtTrig & TrigStore[a]) t_trigFired = true;
+			//LHC16k and LHC16l are CD dedicated runs! CD-online-trigger is used
+			if ( (fEvt->GetFiredTriggerClasses().Contains("CINT7-B-NOPF-CENT")) &&
+				 (fTaskOpt.Contains("LHC16k") || fTaskOpt.Contains("LHC16l")) &&
+				 (TrigStore[a] == AliVEvent::kINT7) ) t_trigFired = true;
+		}
+		if (!t_trigFired) return; //Return 7d
+		else fHisto->FillTH1("EvtCut", "Trig>0", 1); //@
+		const bool t_trigFiredMB   = (fEvtTrig & AliVEvent::kINT7)?true:false;
+		const bool t_trigFiredHMV0 = (fEvtTrig & AliVEvent::kHighMultV0)?true:false;
+
+		fHisto->FillTH1("EvtTrig", "Any", 1); //@
+		if (t_trigFiredMB) fHisto->FillTH1("EvtTrig", "MB", 1); //@
+		if (t_trigFiredHMV0) fHisto->FillTH1("EvtTrig", "HMV0", 1); //@
+
+		//Activate RDHF cut by invoking it, then check pileup status by using multi vertexer
+		const bool t_evtGoodMB   = fEvtCutsMB  ->IsEventSelected(fEvt);
+		const bool t_evtGoodHMV0 = fEvtCutsHMV0->IsEventSelected(fEvt);
+		fEvtPileupMB   = fEvtCutsMB  ->IsEventRejectedDueToPileup();
+		fEvtPileupHMV0 = fEvtCutsHMV0->IsEventRejectedDueToPileup();
+		if (fEvtPileupMB && fEvtPileupHMV0) return; //Return 8d
+		else fHisto->FillTH1("EvtCut", "PileupX>0", 1); //@
+
+		//Check vtx validity (NOT vtxZ)
+		fEvtVtx = fEvt->GetPrimaryVertex();
+		if (!fEvtVtx || fEvtVtx->GetNContributors()<cut_vtxNContributors) return; //Return 9d
+
+		//Store AliNormalizationCounter: DO NOT CHANGE THIS POSITION! (after vtx found, before |VtxZ| < 10)
```

Comment:
Are you sure this is the correct position? Please check wrt other tasks. It should be done before the RDHFCuts->IsEventSelected() call. But since you don't use that in the same way, please make sure it is at the same position

Reply:
This is the function I use later, to retrieve # of events for normalization [link](http://alidoc.cern.ch/AliPhysics/master/_ali_normalization_counter_8cxx_source.html#l00179).

```
Double_t AliNormalizationCounter::GetNEventsForNorm()
{
    Double_t noVtxzGT10=GetSum("noPrimaryV")*GetSum("zvtxGT10")/GetSum("PrimaryV");
    return GetSum("countForNorm")-noVtxzGT10;
}
```

It applies correction for the case "|vtx_z|| > 10, for vertex object not found".
To do that, it estimates the ratio of "vtx object found"  and "vtx object found + |vtx_z| < 10".
If I |vtx_z| < 10 beforehand the estimation screws up, since there will be no "|vtx_z| > 10" yields at all.

Reply:
Since the `StoreEvent()` function does its [own event selection](https://github.com/alisw/AliPhysics/blob/master/PWGHF/vertexingHF/AliNormalizationCounter.cxx#L179) (based on the `AliRDHFCuts` object provided as argument [*]) and fills the relevant histograms based on the outcome of the various event selections. For this to properly work (i.e. you don't reject events in your analysis task that the `AliNormalizationCounter::StoreEvent()` function needs to access for the proper normalisation) it is best to put the call on top before any event selection. There are a few exceptions which aren't checked in the `AliRDHFCuts::IsEventSelected()` call; the placing of the `StoreEvent()` call should therefor be after:
 - the deltaAOD protection ([AliAnalysisTaskSEDs.cxx#L529-L541](https://github.com/alisw/AliPhysics/blob/master/PWGHF/vertexingHF/vHFML/AliAnalysisTaskSEDs.cxx#L529-L541)). Probably not relevant for your task though.
 - checking AOD pointer and TClonesArrays ([AliAnalysisTaskSEDs.cxx#L564-L568](https://github.com/alisw/AliPhysics/blob/master/PWGHF/vertexingHF/vHFML/AliAnalysisTaskSEDs.cxx#L564-L568))
 - checking for null vertex pointer and a non-zero magnetic field ([AliAnalysisTaskSEDs.cxx#L570-L572](https://github.com/alisw/AliPhysics/blob/master/PWGHF/vertexingHF/vHFML/AliAnalysisTaskSEDs.cxx#L570-L572))

In particular, I was doubting your previous placing due to the `IsEventRejectedDueToPileup()` and `GetNContributors()` selections that you placed above the `StoreEvent()` call.

Btw, please refer to https://github.com/alisw/AliPhysics for the latest code, http://alidoc.cern.ch/ is not up to date.

[*] This is another reason why I recommend you to use the IsEventSelected() from the AliRDHFCuts object instead of your own event-selection implementation, to avoid possible mismatches between both event selections leading to the wrong number of estimated events.

Reply:
Done. For its updated position it'd be better to discuss it again in the next pull request.